Sming/Libraries/OtaUpgrade/OTA_FileFormat.h

```diff
@@ -0,0 +1,38 @@
+/****
+ * Sming Framework Project - Open Source framework for high efficiency native ESP8266 development.
+ * Created 2017 by Slavey Karadzhov
+ * http://github.com/SmingHub/Sming
+ * All files of the Sming Core are provided under the LGPL v3 license.
+ *
+ * OTA_FileFormat.h - Definitions for Upgrade file format
+ *
+ * Important: Definitions in this file must be kept in sync with otatool.py
+ * See README.rst for further information.
+ *
+ ****/
+
+#pragma once
+
+#include <stdint.h>
+
+/** File header of an unencrypted OTA upgrade file.
+ */
+typedef struct __attribute__((packed)) {
```

Comment:
My intention was to keep this header C-compatible. I omitted the `extern "C" {` because it contains only types and macros. But I agree, adding `extern "C" {` makes the intent more explicit.

Furthermore: Does Sming offer a macro for `__attribute__((packed))` in a central location? I've seen that you defined it yourself in the FlashString library, so the answer is probably no.

Reply:
As you've defined this structure it doesn't actually require the packed attribute. In this case all the elements are 'naturally aligned' so it can be safely omitted.

Reply:
Another convention for  headers is to use `.hpp` for C++ and `.h` for C, but that is only a convention and doesn't preclude the use of `extern "C"`, etc. I'm not going to propose we adopt it universally, although in some places it makes sense (e.g. FlashString as it's heavily templated).

Reply:
NB. OTA_RomHeader doesn't require packed attribute either.